### DISCUSSION TITLE
Is it a common practice to make one commit mixing the changes with origin codes, when forked other's repository?

### DISCUSSION THREAD

**Discussion by wangzhen11aaa** [databendlabs/databend#3980] [Q&A] [answered]
Hi, all, After reading the codes about hashmap, (curiously) I want to know the changes from hashlink to ritelinked only to find one commt alone. So come with this question. Thanks in advance.

**Comment by PsiACE** [comment, marked as answer]
This issue is not related to Databend.

---

As the maintainer of ritelinked, I don't think the behavior of ritelinked is a common practice. 

The best practice is to maintain the original project, the second best is to fork and keep commits, and the worst is to mix all commits.

For the differences between ritelinked and hashlink, and some additional notes, see: https://github.com/dataslime/ritelinked/issues/6
